docs/extend/modular-resources/create.md

```diff
@@ -0,0 +1,528 @@
+    if len(sys.argv) < 2:
+        raise Exception("Need socket path as command line argument")
+
+    asyncio.run(main(sys.argv[1]))
+
+```
+
+</details>
+
+{{% /tab %}}
+{{% tab name="Go"%}}
+
+<file>main.go</file> is the Go module's entry point file.
+When executed, it initializes the `mybase` custom model and API helper functions from the registry.
+
+<details>
+  <summary>Click to view sample code from <file>main.go</file></summary>
+
+``` go {class="line-numbers linkable-line-numbers"}
+// Package main is a module which serves the mybase custom model.
+package main
+
+import (
+    "context"
+
+    "github.com/edaniels/golog"
+
+    "go.viam.com/rdk/components/base"
+    "go.viam.com/rdk/module"
+    "go.viam.com/utils"
+
+    // NOTE: You must update the following line to import your local package "mybase"
+    "go.viam.com/rdk/examples/customresources/models/mybase"
+)
+
+func main() {
+    // NewLoggerFromArgs will create a golog.Logger at "DebugLevel" if
+    // "--log-level=debug" is an argument in os.Args and at "InfoLevel" otherwise.
+    utils.ContextualMain(mainWithArgs, module.NewLoggerFromArgs("yourmodule"))
+}
+
+func mainWithArgs(ctx context.Context, args []string, logger golog.Logger) (err error) {
+    myMod, err := module.NewModuleFromArgs(ctx, logger)
+    if err != nil {
+        return err
+    }
+
+    // Models and APIs add helpers to the registry during their init().
+    // They can then be added to the module here.
+    err = myMod.AddModelFromRegistry(ctx, base.API, mybase.Model)
+    if err != nil {
+        return err
+    }
+
+    err = myMod.Start(ctx)
+    defer myMod.Close(ctx)
+    if err != nil {
+        return err
+    }
+    <-ctx.Done()
+    return nil
+}
+```
+
+</details>
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{% alert title="Important" color="note" %}}
+
+You must define all functions belonging to a built-in resource subtype's API if defining a new model.
+Otherwise, the class won’t instantiate.
+
+- If you are using the Python SDK, raise an `NotImplementedError()` in the body of functions you do not want to implement or put `pass`.
+- If you are using the Go SDK, return `errUnimplemented`.
+- Additionally, return any values designated in the function's return signature, typed correctly.
+
+{{% /alert %}}
+
+### Compile the module into an executable
+
+To [add a module](/extend/modular-resources/configure/#configure-your-module) to the configuration of your robot, you need to have an [executable](https://en.wikipedia.org/wiki/Executable) that runs your module when executed, can take a local socket as a command line argument, and cleanly exits when sent a termination signal.
+
+Your options for completing this step are flexible, as this file does not need to be in a raw binary format.
+
+{{% tabs %}}
+{{% tab name="Python" %}}
+
+If using the Python SDK, one option is to create and save a new shell script (<file>.sh</file>) that runs your module at your entry point (main program) file.
```

Comment:
```suggestion
Create and save a new shell script (<file>.sh</file>) that runs your module at your entry point (main program) file.
```
They're already on the Python tab so you can take that part out. What is the other option if not this option? Either say what it is or take that out I think.

Reply:
I was working on this section refactor with james quite a bit recently, the diff would just be some words if I hadn't moved the text from a different page.  In my understanding the wording is intentional bc there are a lot of different ways you can set the module executable. This is what James suggested as simplest use case for each sdk